### PR TITLE
Comments in interviews

### LINKED ISSUE #688: Comments in interviews
Possibility to add a comment on an interview

1) Update model
2) Bring comment view from harmony
3) email notification for the HR

### PR DESCRIPTION
**Pull request by Katrin-kudryash**
## PR includes

-   [x] Feature

## Related issues

Resolve #688

### CODE REVIEW

**Comment by Katrin-kudryash** [comment]
https://github.com/taskany-inc/hire/assets/81121562/efdae01c-d66f-4c12-ad3d-3a7147b2f6cb

File: src/components/Interview/Interview.tsx [hunk 1]
```diff
@@ -35,13 +40,34 @@ interface InterviewProps {
     rejectReasons: RejectReason[];
 }
 
+const StyledTitle = styled(Text)`
+    margin-top: 50px;
+    display: flex;
+    margin-bottom: 40px;
+    cursor: pointer;
```
Comment (tsumo) [inline src/components/Interview/Interview.tsx:47]: Very strange styles for title. Is it copypasted?

File: src/components/Interview/Interview.tsx [hunk 2]
```diff
@@ -35,13 +40,34 @@ interface InterviewProps {
...
+    align-items: center;
+    gap: 5px;
+`;
+
+const StyledComment = styled.div`
+    display: grid;
+    grid-template-column: 7fr 5fr;
```
Comment (tsumo) [inline src/components/Interview/Interview.tsx:54]: Same here. Why columns? Why flex-wrap?

File: src/components/EditOrDeleteComment/EditOrDeleteComment.tsx [hunk 3]
```diff
@@ -4,11 +4,11 @@ import { useCommentDeleteMutation, useCommentEditMutation } from '../../modules/
 import { CommentSchema, CommentWithUser } from '../../modules/commentTypes';
 import { CommentView } from '../CommentView/CommentView';
 
-interface ProblemCommentsProps {
+interface EditOrDeleteCommentProps {
     comment: CommentWithUser;
 }
 
-export const ProblemComment = ({ comment }: ProblemCommentsProps) => {
+export const EditOrDeleteComment = ({ comment }: EditOrDeleteCommentProps) => {
```
Comment (tsumo) [inline src/components/EditOrDeleteComment/EditOrDeleteComment.tsx:11]: Just "Comment" would be a better name

File: src/modules/commentMethods.ts [hunk 4]
```diff
@@ -9,25 +9,58 @@ import { sendMail } from './nodemailer';
...
     });
 
-    if (comment.user !== comment.problem.author) {
-        return sendMail({
+    if (comment.user !== comment.problem?.author && comment.problem?.author) {
+        await sendMail({
             to: comment.problem.author.email,
-            subject: tr('Comment on proplem {comment.problem.name}'),
+            subject: `${tr('Comment on problem')} ${comment.problem.name}`,
             text: `User ${comment.user.name} wrote:\n
```
Comment (tsumo) [inline src/modules/commentMethods.ts:42]: Missing translation

**Review by tsumo — CHANGES_REQUESTED** [review]

File: src/components/Interview/Interview.tsx (hunk 1 above)
Comment (Katrin-kudryash) [reply src/components/Interview/Interview.tsx]: oh, yes, I deleted it

File: src/modules/commentMethods.ts [hunk 5]
```diff
@@ -9,25 +9,58 @@ import { sendMail } from './nodemailer';
...
+                problemId: comment.problem.id,
             })}`,
         });
     }
+    if (comment.user !== comment.interview?.creator && comment.interview?.creator) {
+        if (comment.interview?.creator) {
+            await sendMail({
+                to: comment.interview.creator.email,
+                subject: `${tr('Comment on interview')} ${comment.interview.id}`,
+                text: `User ${comment.user.name} wrote:\n
```
Comment (tsumo) [inline src/modules/commentMethods.ts:54]: Missing translation

**Review by tsumo — CHANGES_REQUESTED** [review]

File: src/modules/commentMethods.ts (hunk 5 above)
Comment (Katrin-kudryash) [reply src/modules/commentMethods.ts]: done

**Review by tsumo — APPROVED** [review]

**Review by 9teen90nine — APPROVED** [review]

**Pull request merged**